### PR TITLE
plat/linuxu: Reorder macros in the linuxu system call header files

### PR DESCRIPTION
**Pull request by flpostolache**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): linuxu
 - Application(s): N/A

### Description of changes

This commit reorders all system call macros in the
/plat/linuxu/include/linuxu header files after their system call numbers.

### CODE REVIEW

**Comment by nderjung** [comment]
I wonder if these should also be double-tab aligned for shorter SC names, what do you think @skuenzer?
```diff
-#define __SC_EXIT      1
-#define __SC_READ      3
-#define __SC_WRITE     4
-#define __SC_CLOSE     6
-#define __SC_IOCTL     54
-#define __SC_FCNTL     55
-#define __SC_MUNMAP    91
-#define __SC_FSTAT     108
+#define __SC_EXIT              1
+#define __SC_READ              3
+#define __SC_WRITE             4
+#define __SC_CLOSE             6
+#define __SC_IOCTL             54
+#define __SC_FCNTL             55
+#define __SC_MUNMAP            91
+#define __SC_FSTAT             108
 #define __SC_RT_SIGPROCMASK    126
 #define __SC_RT_SIGACTION      174
-#define __SC_MMAP      192 /* use mmap2() since mmap() is obsolete */
+#define __SC_MMAP              192 /* use mmap2() since mmap() is obsolete */
 #define __SC_TIMER_CREATE      257
 #define __SC_TIMER_SETTIME     258
 #define __SC_TIMER_GETTIME     259
 #define __SC_TIMER_GETOVERRUN  260
 #define __SC_TIMER_DELETE      261
 #define __SC_CLOCK_GETTIME     263
-#define __SC_SOCKET    281
-#define __SC_OPENAT    322
-#define __SC_PSELECT6  335
+#define __SC_SOCKET            281
+#define __SC_OPENAT            322
+#define __SC_PSELECT6          335
```

**Comment by flpostolache** [comment]
@nderjung That was my first option too. But after that, I saw there was a lot of space between the short name macros and their corresponding number and it was kinda hard to follow. I will wait for your comment and correct it accordingly.

**Comment by razvand** [comment]
@skuenzer , what do you think? Should we integrate this PR or discard it?

**Comment by skuenzer** [comment]
> @skuenzer , what do you think? Should we integrate this PR or discard it?

Usually, I am not a big fan of PRs that re-order definitions in headers because they may easily break other outstanding PRs. On the other hand I think we are still safe with `plat/linuxu`, so this should be fine. I guess this PR should be rebased, but I am okay if we take it in.

**Review by razvand — APPROVED** [review]
All OK.

Reviewed-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

**Comment by razvand** [comment]
@nderjung , regarding the indentation, I would keep this as a topic for a future code conventions discussion.

**Review by skuenzer — APPROVED** [review]
Approved-by: Simon Kuenzer <simon.kuenzer@neclab.eu>
